internal/configs/testdata/valid-files/output-type-constraint.tf

```diff
@@ -0,0 +1,11 @@
+output "string" {
+  type  = string
+  value = "Hello"
+}
+
+output "object" {
+  type = object({
+    name = optional(string, "Bart"),
+  })
+  value = {}
+}
```

Comment:
Should we support both explicit and implicit default values?

This already introduces support for implicit defaults through optional attributes. While I initially thought against defaults entirely, they likely provide real value to users. For consistency sake,
Perhaps we should support both patterns rather than just one.

Reply:
I'm not sure if I understand what you're asking here. How would an explicit default value work in the context of an output block?

Reply:
I was thinking we could assign the default when the assigned attribute is null, 
like we have for variables https://developer.hashicorp.com/terraform/language/values/variables#default-values

Reply:
Does this mean you're proposing to introduce a new `default` attribute for output blocks? I think we would have to discuss whether that makes sense from a language point of view.

Unlike variables, outputs usually get their values from configuration and don't depend on user-supplied values. Where a user might only supply some of the variables, but not all, we fall back on the default. An output should always have a value, even if it is sometimes `null`.

The `optional` modifier in the above test case only works within objects https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes

Reply:
Yes, I was proposing a new `default` attribute for outputs. Your explanation is very helpful, thank you.

> An output should always have a value, even if it is sometimes null.

Make sense - since outputs are guaranteed to have a value (with `null` being a valid value), then there's no need for default handling.

When this is being documented, I think it would however make sense to add some documentation clarifying this distinction between variables and outputs, particularly around null handling? This might help other users avoid similar misconceptions.